liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/ServerXmlDocument.java

```diff
@@ -42,38 +45,84 @@
     
     private ServerXmlDocument() {    
     }
+
+    /**
+     * @param filePath
+     * @param fileName application file name
+     * @return true if the application with the file name is configured.
+     * @throws Exception
+     */
+    private static boolean isFoundWebAppWithFileName(String filePath, String fileName) throws Exception {
+        return isFoundAppConfigByTagAndFileName(filePath, "webApplication", fileName);
+    }
     
-    public static boolean isFoundTagName(String fileName, String tag) throws Exception {
+    /**
+     * @param filePath 
+     * @param tag specify element tag to parse
+     * @param fileName application file name
+     * @return true if the application with the file name within the specified element tag is configured. 
+     * @throws Exception
+     */
+    private static boolean isFoundAppConfigByTagAndFileName(String filePath, String tag, String fileName) throws Exception {
 
-        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
-        DocumentBuilder builder = factory.newDocumentBuilder();
-        Document doc = builder.parse(fileName);
+        NodeList appList = getNodeList(filePath, "/server/" + tag);
+        
+        if (appList == null || appList.getLength() == 0) {
+            return false; 
+        }
+        else {
+          if (fileName == null) {
```

Comment:
One use case is, when 'appsDirectory' is 'dropins' and needs a quick check to see if app is configured. If then throw exception. In this case, it doesn't need to check file name. Another use case is quickly return appsDirectory('dropins' or 'apps') by checking if app is configured without considering file name.